src/pages/workspace/workflows/ToggleSettingsOptionRow.tsx

```diff
@@ -29,11 +29,19 @@ const ICON_SIZE = 48;
 function ToggleSettingOptionRow({icon, title, subtitle, onToggle, subMenuItems, isActive, pendingAction}: ToggleSettingOptionRowProps) {
     const [isEnabled, setIsEnabled] = useState(isActive);
     const styles = useThemeStyles();
+
     const toggleSwitch = () => {
-        setIsEnabled(!isEnabled);
-        onToggle(!isEnabled);
+        setIsEnabled((prev) => {
+            onToggle(!prev);
+
+            return !prev;
+        });
     };
 
+    useEffect(() => {
+        setIsEnabled(isActive);
```

Comment:
can't we use isActive and update isActive here?
Without internal state?

Reply:
Done - e9594bc.